### PR TITLE
[tool] Allow running from anywhere

### PR DESCRIPTION
**Pull request by stuartmorgan**
Now that the repo tooling is always run from source, not via `pub global`, we no longer need to infer the repo location from the current directory. Instead, hard-code knowledge of where the repository root is. This makes it much easier to run the tooling, since it's common to be in a package directory rather than the repo root.

To make it even easier to run from within a package, this also adds a `--current-package` as an alternative to `--packages`. This makes it possible to, e.g., write local wrapper scripts that run a specific set of commands on whatever the current package happens to be (such as a generic version of the script discussed in https://github.com/flutter/packages/pull/4129).

As related cleanup, makes the tool non-publishable (we haven't been publishing it since the repo merge, but I never made it unpublishable; this is important now that it would not work if published) and remove the LICENSE and CHANGELOG since it's no longer a stand-alone package.

Fixes https://github.com/flutter/flutter/issues/128231
Fixes https://github.com/flutter/flutter/issues/128232

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/tool/lib/src/common/package_command.dart [hunk 1]
```diff
@@ -545,6 +572,32 @@ abstract class PackageCommand extends Command<void> {
...
 
+  String? _getCurrentDirectoryPackageName() {
+    // Ensure that the current directory is within the packages directory.
+    final Directory absolutePackagesDir = packagesDir.absolute;
+    Directory currentDir = packagesDir.fileSystem.currentDirectory.absolute;
+    if (!currentDir.path.startsWith(absolutePackagesDir.path) ||
+        currentDir.path == packagesDir.path) {
+      return null;
+    }
+    // If the current directory is a direct subdirectory of the packages
```
Comment (tarrinneal) [inline script/tool/lib/src/common/package_command.dart:583]: sub-directory

File: script/tool/README.md [hunk 2]
```diff
@@ -42,31 +38,36 @@ command is targetting. An package name can be any of:
 - A combination federated_plugin_name/package_name (e.g.,
   `path_provider/path_provider` for the app-facing package).
 
+The examples below assume they are being run from the repository root, but
+the script works from anywhere. If you develop in flutter/packages frequently,
```
Comment (tarrinneal) [inline script/tool/README.md:42]: anywhere within packages?

File: script/tool/lib/src/common/package_command.dart [hunk 3]
```diff
@@ -545,6 +572,32 @@ abstract class PackageCommand extends Command<void> {
...
+    // diretory, then that's the target.
```
Comment (tarrinneal) [inline script/tool/lib/src/common/package_command.dart:584]: directory typo

File: script/tool/lib/src/common/package_command.dart (hunk 1 above)
Comment (stuartmorgan) [reply script/tool/lib/src/common/package_command.dart:583]: "subdirectory" is a word (and according to https://www.merriam-webster.com/dictionary/subdirectory is the more common form).

File: script/tool/README.md (hunk 2 above)
Comment (stuartmorgan) [reply script/tool/README.md:42]: Anywhere! `--current-package` only works in the packages directory, but with this change if you want to run `dart run /absolute/path/to/the/tool some-command --package=a_package` from some random other location it'll work.

File: script/tool/lib/src/common/package_command.dart (hunk 1 above)
Comment (tarrinneal) [reply script/tool/lib/src/common/package_command.dart:583]: hmm, can't argue with that. Seems a little strange to me.

Comment (stuartmorgan) [reply script/tool/lib/src/common/package_command.dart:583]: I was trying to think of other examples, and it seems like sub is mostly not hyphenated. I'm not sure if there's a rule, or it's one of those things that gets dropped in anything where it's used frequently.

Comment (tarrinneal) [reply script/tool/lib/src/common/package_command.dart:583]: subbasement, subfloor... I'm just wrong I guess.

**Pull request merged**